commit 0c6f1f4ea4980ff719979d36f10bd6ea3e464c02
Author: Jan Janssen <medhefgo@web.de>
Date:   Thu Oct 31 17:22:03 2013 +0100

    Make hibernation test work for swap files
    
    Suspend to disk works for swap files too (even if it is located
    on an ecrypted file system):
    https://www.kernel.org/doc/Documentation/power/swsusp-and-swap-files.txt

diff --git a/src/shared/sleep-config.c b/src/shared/sleep-config.c
index d068bfc..2bb0493 100644
--- a/src/shared/sleep-config.c
+++ b/src/shared/sleep-config.c
@@ -206,7 +206,7 @@ static int hibernation_partition_size(size_t *size, size_t *used) {
                 if (!d)
                         return -ENOMEM;
 
-                if (!streq(type, "partition")) {
+                if (!streq(type, "partition") && !streq(type, "file")) {
                         log_debug("Partition %s has type %s, ignoring.", d, type);
                         continue;
                 }
